### PR TITLE
Change replaceInnerBlocks updateSelection property to false

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Based on this comment: https://github.com/WordPress/gutenberg/pull/26267#pullrequestreview-511686901

This PR changes the default value of action `replaceInnerBlocks`'s `updateSelection` property to `false`. This change propagates to changing the default value of `InnerBlock`'s `templateInsertUpdatesSelection` to `false` as well.

This change affects whether the selection is updated or not, when child blocks in the template are inserted.

Most blocks that use `InnerBlocks` would set this as `false`, thus making this change the preferred default value.

 

### Affected blocks:
- Query
- Columns
- Column
- Navigation
- Media & Text
- Buttons
- Cover

### Hooks
- useBlockSync

### Native files/Blocks/Components changed
- Media & Text
- BlockVariationPicker
- Columns
- Buttons
- Cover

You can test this by creating the above affected blocks on master and the on this branch. Block `focus` should be identical.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/store/actions.js [hunk 1]
```diff
@@ -668,7 +668,7 @@ export function removeBlock( clientId, selectPrevious ) {
 export function replaceInnerBlocks(
 	rootClientId,
 	blocks,
-	updateSelection = true
+	updateSelection = false
```
Comment (gziolo) [inline packages/block-editor/src/store/actions.js:671]: JSDoc needs to be updated.

**Review by gziolo — COMMENTED** [review]
One of the plugins used with e2e tests need to be updated to ensure that old behavior is preserved:

https://github.com/WordPress/gutenberg/blob/9ce9380681c5c02caed074fbe1539b2174b18319/packages/e2e-tests/plugins/inner-blocks-templates/index.js#L67

`templateInsertUpdatesSelection` should be set to `true` there.


It feels like it's a good direction. In most of the cases, the changes applied, they align with the most popular behavior.

File: packages/block-library/src/cover/edit.native.js [hunk 2]
```diff
@@ -514,7 +514,10 @@ const Cover = ( {
 				pointerEvents="box-none"
 				style={ [ styles.content, { minHeight: CONTAINER_HEIGHT } ] }
 			>
-				<InnerBlocks template={ INNER_BLOCKS_TEMPLATE } />
+				<InnerBlocks
+					template={ INNER_BLOCKS_TEMPLATE }
+					templateInsertUpdatesSelection={ true }
```
Comment (gziolo) [inline packages/block-library/src/cover/edit.native.js:519]: It should be equivalent with:
```suggestion
					templateInsertUpdatesSelection
```

**Comment by guarani** [comment]
@ntsekouras thanks for the heads-up about the `*.native.js` changes. I smoke tested Media & Text, Columns, Buttons and Cover on mobile and everything looks OK 👍. The changes affecting native files appear to be only due to `templateInsertUpdatesSelection` previously defaulting to `true` and now defaulting to `false`.

**Comment by gziolo** [comment]
It looks like this PR needs to be rebased.

I also marked it as needing a short dev note for WordPress 5.7 release. Can you compile something upfront so the process is more streamlined?

**Comment by ntsekouras** [comment]
Dev Note for this change.

### The default value of `replaceInnerBlocks` action's `updateSelection` property is now set to `false`.

This change propagates to changing the default value of `InnerBlock`'s `templateInsertUpdatesSelection` to false as well because `templateInsertUpdatesSelection` is passed in the call of `replaceInnerBlocks`.

This change affects whether the selection is updated or not, when:
- we use `replaceInnerBlocks` action to replace a block's InnerBlocks
- child blocks in the `template` property are inserted in a InnerBlocks block.

Most blocks that use `InnerBlocks` would set this as `false`, thus making this change the preferred default value.

#### Some examples

Before:
```
// We didn't want to update the selection to the new InnerBlocks,
// so the selected block would remain the parent block.
replaceInnerBlocks( clientId, controlledBlocks, false );
```
Now: 
```
replaceInnerBlocks( clientId, controlledBlocks );
```

```
// This will update the selection to the first new block created from the template `INNER_BLOCKS_TEMPLATE`.
const innerBlocksProps = useInnerBlocksProps(
	{       className: 'wp-block-cover__inner-container'       },
	{
		template: INNER_BLOCKS_TEMPLATE,
		templateInsertUpdatesSelection: true,
	}
);
```

**Comment by gziolo** [comment]
@ntsekouras, how do you feel about using the following intro to this dev note that explains the reasoning behind these changes:

## Inner Blocks API Changes

We noticed that the majority of block authors prefer to keep the focus on the parent block upon insertion in the editor rather than move it to one of the child blocks. Prior to the changes introduced the default behavior was the latter, so we decided to change it to follow the more popular choice that is going to simplify API usage. To accomplish that we had to change the definition of the following parts of API:
- `InnerBlock` component
- `useInnerBlocksProps` hook
- `replaceInnerBlocks` action

**Review by gziolo — APPROVED** [review]
This looks good, thank you for all the work on these changes.

**Pull request merged**